docs/evm_compatibility/overview.md

```diff
@@ -1,63 +1,155 @@
+
+- Basic accounts store an account balance
+- Code accounts store EVM code
+
+Solana’s Sealevel also provides two accounts which are assigned up to 256 consecutive storage slots; but they have different purposes: 
+
+ - Executable
+
+ > Executable accounts provide immutable storage of executable byte code or an (immutable) proxy address of an account which stores mutable executable byte code.
+
+ - Non-executable
+
+> Since executable accounts are immutable, their application state is stored in non-executable accounts.
+
+The differences go deeper still. In an Ethereum EVM, contracts can only read and write their own storage. In Sealevel, any account’s data can be read or written to by a contract. However, the runtime enforces that only an account’s “owner” is allowed to modify it. Changes by any other programs will be reverted and cause the transaction to fail.
+
+:::info
+
+Learn more in the [Solana wiki](https://solana.wiki/docs/).
+
+:::
+
+### Account mapping
+
+Each Ethereum account involved in a transaction must be mapped to a corresponding Solana account. Any call made to the Ethereum account (e.g. to read balance, execute a transaction, etc.) requires that the Solana account is included to make use of the storage provided.
 
-### Upper limit on number of Accounts
-Neon EVM uses Solana Transaction V0 with an extension via [account lookup tables](https://docs.solana.com/developing/lookup-tables): this limits the maximum number of accounts to 64 in a single transaction. Solana requires that all accounts used in a transaction be specified in order to ensure parallel execution of transactions. 
+<!-- based on item in Slack https://neonlabsworkspace.slack.com/archives/C03CQ8A6WTT/p1683107335962669 >> not sure that this is functional end user support as it stands -->
+
+When a contract requires storage slots within Solana, it may create Solana accounts and access these random addresses. However, this must be done with an awareness of the upper limit on the number of accounts, as per the next section. 
+
+
+### Upper limit on number of accounts
+Neon EVM uses [Solana Transaction V0](https://docs.solana.com/developing/versioned-transactions): limiting the maximum number of accounts used in a single transaction to 64. Solana requires that all accounts used in a transaction be specified in order; to ensure parallel execution of transactions.
+
+<!-- go deeper on HOW to modify the contract to constrain account numbers Anton will pass in slack
+  -->
+
+By constructing the contract logic differently, fixed-sized values and arrays can fit into a significantly smaller number of accounts.
+
+```Solidity
+ /*1 account: */
+uint256[] list;
+...
+
+for (uint256 i = 0; i < 32; i++) {
+   list[i] = i;
+}
+```
+
+```Solidity
+/* 32 accounts: */
+mapping(uint256 => uint256) map;
+...
+
+for (uint256 i = 0; i < 32; i++) {
+   map[i] = i;
+}
+```
+
+<!-- todo looks like these code snippets will benefit from some better commentary -->
```

Comment:
@anton-lisanin can you provide deeper information for comments in code -- or a deeer explanation of the scenario to which these 2 variants may be applied to?
Just a bit more context may assist the reader.

Reply:
Code is extremely simple. There really nothing to comment.
Context was lost when moving from Slack

Reply:
@anton-lisanin We need just 1 sentence describing the difference between the first and second examples -- why one is preferable to the other @YYYurchenko agrees we need a bit of context here.

Reply:
@afalaleev can we have a few lines on why map good list bad (if that is the correct logic?!)?